Revise caching/queueing configs and docs for Redis locking (#319)
nautobot/nautobot: nautobot/docs/configuration/required-settings.md

> For more details Nautobot's caching see the guide on [Caching](../../additional-features/caching).

Reviewer: Are any changes needed in the `additional-features/caching.md` document (which I admit I've never read before)?

Author: I did add a small admonition there, but other than that I don't think anything else needs to be added.

!!! important

>     Cacheops does not utilize the built-in [Django cache framework](https://docs.djangoproject.com/en/stable/topics/cache/) to perform caching. Instead it [monkey patches](./../additional-features/caching.md#django_cacheops) the underlying queryset methods to intercept calls to get and set cached items in Redis.

Reviewer: ```suggestion
    Nautobot does not utilize the built-in [Django cache framework](https://docs.djangoproject.com/en/stable/topics/cache/) to perform caching, as `django-cacheops` takes its place.
```

Default: `undefined`

If you are using [Redis Sentinel](https://redis.io/topics/sentinel) for high-availability purposes, you must replace the [`CACHEOPS_REDIS`](#cacheops_redis) setting with [`CACHEOPS_SENTINEL`](#cacheops_sentinel).

!!! warning

>     [`CACHEOPS_REDIS`](#cacheops_redis) and [`CACHEOPS_SENTINEL`](#cacheops_sentinel) are mutually exclusive and will result in an error if both are set.

Reviewer: Is this a built-in error check, or something we've added?

Author: Built-in. Didn't need to write a check for it.

### Task Queuing

Task queues are configured by defining them within the [`RQ_QUEUES`](#rq_queues) setting. 

Nautobot's core functionality relies on several distinct queues and these represent the minimum required set of queues that must be defined. By default, these use identical connection settings as defined in [`CACHES`](#caches) (yes, that's confusing and we'll explain below).

> It is up to you to modify the task queues for your environment and know that other use cases such as utilizing specific plugins may require additional queues to be defined. In most cases the default settings will be suitable for production use.

Reviewer: ```suggestion
In most cases the default settings will be suitable for production use, but it is up to you to modify the task queues for your environment and know that other use cases such as utilizing specific plugins may require additional queues to be defined. 
```

#### CACHES

> The [`django-redis`](https://github.com/jazzband/django-redis) Django plugin is used to provide the backend for Redis as a concurrent write lock for preventing race conditions when allocating IP address objects and to define centralized Redis connection settings that will be used by RQ. This setting is used to to simplify the configuration for defining queues. *It is not used for caching at this time.* 

Reviewer: ```suggestion
The [`django-redis`](https://github.com/jazzband/django-redis) Django plugin is used to provide the backend for Redis as a concurrent write lock for preventing race conditions when allocating IP address objects and to define centralized Redis connection settings that will be used by RQ. The `CACHES` setting is used to to simplify the configuration for defining queues. *It is not used for caching at this time.* 
```

> The [`django-redis`](https://github.com/jazzband/django-redis) Django plugin is used to provide the backend for Redis as a concurrent write lock for preventing race conditions when allocating IP address objects and to define centralized Redis connection settings that will be used by RQ. The `CACHES` setting is required to to simplify the configuration for defining queues. *It is not used for caching at this time.*

Reviewer: ```suggestion
The [`django-redis`](https://github.com/jazzband/django-redis) Django plugin is used to enable Redis as a concurrent write lock for preventing race conditions when allocating IP address objects, and also to define centralized Redis connection settings that will be used by RQ. The `CACHES` setting is required to to simplify the configuration for defining queues. *It is not used for caching at this time.*
```

!!! important

>     Nautobot does not utilize the built-in [Django cache framework](https://docs.djangoproject.com/en/stable/topics/cache/) to perform caching because Cacheops is being used instead as detailed just above. *Yes, we know this is confusing, which is why this is being called out explicitly!*

Reviewer: ```suggestion
    Nautobot does not utilize the built-in [Django cache framework](https://docs.djangoproject.com/en/stable/topics/cache/) (which also relies on the `CACHES` setting) to perform caching because Cacheops is being used instead as detailed just above. *Yes, we know this is confusing, which is why this is being called out explicitly!*
```
